### ISSUE TITLE
support multiple host_network names for the same interface

### ISSUE THREAD

**Issue by AndrewChubatiuk** [hashicorp/nomad#10106]
### Nomad version
nomad v1.0.4

### Issue
Having issue with multi-region infrastructure setup. One region is in AWS VPC and another is in Hetzner + [Nebula](https://github.com/slackhq/nebula) for private networks setup. Host networks configuration in Hetzner is:
```
{
  "host_network": {
    "private": {
      "cidr": "<nebula-ip-address>/32"
    },
    "public": {
      "cidr": "<hetzner-ip-address>/32"
    }
  }
}
```

To be able to use the same job for both regions (cause there's no `host_network` variable interpolation) had to use the same interface for both public and private agent networks configuration in AWS for compatibility
```
{
  "host_network": {
    "private": {
      "cidr": "<awsvpc-ip-address>/32"
    },
    "public": {
      "cidr": "<awsvpc-ip-address>/32"
    }
  }
}
```
But nomad doesn't create allocations in AWS a job, which uses both public and private interfaces, cause nomad agent is registred with `private` network only. Is there any option to either add `host_network` interpolation support or use multiple network names for the same interface?

**Comment by tgross** [comment]
Hi @AndrewChubatiuk! 

> But nomad doesn't create allocations in AWS a job, which uses both public and private interfaces, cause nomad agent is registred with private network only. 

I'm not sure I understand... can you explain the behavior you're seeing here a bit more?

**Comment by AndrewChubatiuk** [comment]
@tgross 
I have a problem when trying to apply the same job for a multiregion setup with different host_network configurations. In AWS region we have private network interface only with AWS VPC ip address and in a region that is in Hetzner we have both private (Nebula) and public network interfaces. To apply the same job for both regions there should be either ability to interpolate host_network value in port stanza (I could add host_network name to nomad client meta) or specify multiple host_hetwork aliases for the same interface (I could give multiple names to an AWS VPC host_network and in this case not host_network interpolation needed)

**Comment by groggemans** [comment]
I'm also running into this issue, but in a slightly different way.

On several of my test nodes I have a large amount of host networks. To properly identify them they are all configured as `host_network` in the agent config. As a result I also set a fixed name for what otherwise would be the "default" network range.

However that default range is different between some of the nodes. As a result I have no way to schedule a job in the "default" network, as that is no longer known to Nomad. Using the `host_network` name instead would result in the problem outlined by Andrew above.

For my use case there would be another solution besides those mentioned by Andrew; And that is to let Nomad also define the "default" host network, even if a `host_network` with the same range is defined. This would also work in Andrews case, but I do agree that defining the same range with multiple names or interpolating the name would be cleaner for that use case.

Losing the "default" range when making all host networks explicit also results in jobs being unable to schedule if no explicit `host_network` is defined for a port (also for connect sidecars). An inconvenience for me, but could be used to enforce more explicit network definitions for others. Multiple aliases or interpolation is probable the better solution as it still suits both behaviors.

**Comment by tgross** [comment]
Closed by #10104, which will ship in Nomad 1.1.0
